networkit/cpp/graph/Graph.cpp

```diff
@@ -1065,4 +1068,52 @@ Graph Graph::subgraphFromNodes(const std::unordered_set<node> &nodes) const {
 	return S;
 }
 
+void Graph::toLargestConnectedComponent() {
+	if (this->isDirected())
+		throw std::runtime_error(
+			"Method available for undirected graphs only.");
+
+	ConnectedComponents cc(*this);
+	cc.run();
+	auto comps = cc.getComponents();
+	std::sort(comps.begin(), comps.end(),
+		[&](const std::vector<node> &x, const std::vector<node> &y) {
+			return x.size() > y.size();
+	});
+	std::unordered_map<node, node> map;
+	std::unordered_set<node> set(comps[0].begin(), comps[0].end());
+	node idx = 0;
+	for (node u : comps[0])
+		map[u] = idx++;
```

Comment:
Why are you using an unordered map here instead of a simple `std::vector` over all nodes which should give a much better running time? We have O(n) storage used anyway (note that the `subgraphFromNodes` below even copies the whole graph).